src/algorithms/synthesis/dd_synthesis.cpp

```diff
@@ -52,4 +54,350 @@ namespace syrec {
         const auto label = static_cast<dd::Qubit>(tt.nInputs() - 1U);
         return dd->makeDDNode(label, edges);
     }
+
+    auto DDSynthesis::pathFromRoot(dd::mNode* src, dd::mNode* dst, TruthTable::Cube::Vector& sigVec, TruthTable::Cube& tempVec) -> void {
+        if (bool bitMatch = (src->v > dst->v); !bitMatch) {
+            if (src == dst) {
+                sigVec.push_back(tempVec);
+                return;
+            }
+            return;
+        }
+
+        auto* n = src->e[0].p;
+        tempVec.emplace_back(false);
+        pathFromRoot(n, dst, sigVec, tempVec);
+        tempVec.pop_back();
+
+        auto* p = src->e[3].p;
+        tempVec.emplace_back(true);
+        pathFromRoot(p, dst, sigVec, tempVec);
+        tempVec.pop_back();
+        //return
+    }
+
+    auto DDSynthesis::pathSignature(dd::mNode* src, TruthTable::Cube::Vector& sigVec, TruthTable::Cube& tempVec) -> void {
+        if (src->v == 0) {
+            if (src->e[0] == dd::mEdge::one) {
+                tempVec.emplace_back(false);
+                sigVec.push_back(tempVec);
+                tempVec.pop_back();
+            }
+            if (src->e[1] == dd::mEdge::one) {
+                tempVec.emplace_back(true);
+                sigVec.push_back(tempVec);
+                tempVec.pop_back();
+            }
+            if (src->e[2] == dd::mEdge::one) {
+                tempVec.emplace_back(false);
+                sigVec.push_back(tempVec);
+                tempVec.pop_back();
+            }
+            if (src->e[3] == dd::mEdge::one) {
+                tempVec.emplace_back(true);
+                sigVec.push_back(tempVec);
+                tempVec.pop_back();
+            }
+
+            return;
+        }
+
+        if (dd::mNode::isTerminal(src)) {
+            return;
+        }
+
+        auto* p1 = src->e[0].p;
+        tempVec.emplace_back(false);
+        pathSignature(p1, sigVec, tempVec);
+        tempVec.pop_back();
+
+        auto* p2 = src->e[1].p;
+        tempVec.emplace_back(true);
+        pathSignature(p2, sigVec, tempVec);
+        tempVec.pop_back();
+
+        auto* p3 = src->e[2].p;
+        tempVec.emplace_back(false);
+        pathSignature(p3, sigVec, tempVec);
+        tempVec.pop_back();
+
+        auto* p4 = src->e[3].p;
+        tempVec.emplace_back(true);
+        pathSignature(p4, sigVec, tempVec);
+        tempVec.pop_back();
+        //return
+    }
+
+    auto DDSynthesis::unifyPath(dd::mEdge& src, dd::mNode* current, syrec::TruthTable::Cube::Vector& p1SigVec, syrec::TruthTable::Cube::Vector& p2SigVec, const std::vector<std::size_t>& indices, std::unique_ptr<dd::Package<>>& dd) -> void {
```

Comment:
You are already in the `syrec` namespace. So please remove all the `syrec::` namespace specifiers through functions in this file in order to increase readability

Reply:
done 👍🏻